course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ExtendedContentConsumptionActor.scala

```diff
@@ -0,0 +1,798 @@
+package org.sunbird.enrolments
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.apache.commons.collections4.{CollectionUtils, MapUtils}
+import org.apache.commons.lang3.StringUtils
+import org.sunbird.cassandra.CassandraOperation
+import org.sunbird.common.CassandraUtil
+import org.sunbird.common.exception.ProjectCommonException
+import org.sunbird.common.models.response.Response
+import org.sunbird.common.models.util._
+import org.sunbird.common.request.{Request, RequestContext}
+import org.sunbird.common.responsecode.ResponseCode
+import org.sunbird.common.util.JsonUtil
+import org.sunbird.helper.ServiceFactory
+import org.sunbird.kafka.client.{InstructionEventGenerator, KafkaClient}
+import org.sunbird.learner.constants.{CourseJsonKey, InstructionEvent}
+import org.sunbird.learner.util.{ContentCacheHandlerV2, ContentUtil, Util}
+
+import java.time.{ZoneId, ZonedDateTime}
+import java.util
+import java.util.{Date, TimeZone, UUID}
+import javax.inject.Inject
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+
+case class ExtendedInternalContentConsumption(courseId: String, batchId: String, contentId: String) {
```

Comment:
this class contains the entire code from previous one -- few of the methods were unnecessary... 
need to make sure only required methods were added